subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java

```diff
@@ -96,6 +112,15 @@ private PublishSpec computeSpec() {
             throw new InvalidUserDataException("The 'repository' property is required");
         }
 
+        Property<Credentials> credentialsProperty = repository.getConfiguredCredentials();
+        Credentials value = credentialsProperty.getOrNull();
+
+        boolean safeCredentials = value == null || areCredentialsSafe(repository.getName(), value);
+
+        if (!safeCredentials && getServices().get(BuildModelParameters.class).isConfigurationCache()) {
+            issueReporter.reportIssue(ProviderIssue.UNSAFE_CREDENTIALS, MessageFormat.format("Credential values found in configuration for repository {0}", repository.getName()));
```

Comment:
ℹ️ In general, we want to leave UI concerns (like message formatting) for the listener on the configuration cache side to handle since it can better deal with the different presentations (console versus html report).

Reply:
Do you mean, it is preferred to use string concatenation over MessageFormat (which I use in Java as a replacement for lack of string interpolation).

Reply:
It would be preferred to send all the data to the other side so the message can be assembled there, e.g., `onUnsafeCredentials(..., String repositoryName)`.